modules/instagram/classes/controller/manage.php

```diff
@@ -117,25 +117,8 @@ public function action_index()
 
 		$view = \View::forge('manage');
 		$view->set('fieldset', \Propeller\Instagram\Subscription::fieldset(), false);
-		$subscriptions = \Propeller\Instagram\Subscription::get();
-		$prop_sub = \Propeller\Instagram\Model_Subscription::find('all');
-		$merged = array();
-
-		if($prop_sub) {
-			foreach($prop_sub as $subscription) {
-				$merged[$subscription->object_id] = $subscription;
-			}
-
-			foreach($subscriptions as $subscription) {
-				$key = str_replace('#', '', $subscription->object_id);
-				if(isset($merged[$key])) {
-					$merged[$key]->instagram = $subscription;
-				}
-			}
-		}
-
-		$view->set('subscriptions', $merged);
-
+		$subscriptions = \Propeller\Instagram\Model_Subscription::find('all');
```

Comment:
I'd prefer to see repository here, see single responsibility principle. ORM provides nice set of helpers, but it makes it difficult to write unit tests for it.

Reply:
This is a change to existing functionality, only fetched subscriptions are listed. It means user need to wait for cronjob to fetch images to see them here. Some clients might not want to wait.
Is it because new API does not allow it?